OPENAFS raises an EACCES error when trying to set the SUID bits.
This patch helps fakeroot with this case.
--- fakeroot-1.36/libfakeroot.c.orig	2025-04-28 21:14:11.114169874 -0700
+++ fakeroot-1.36/libfakeroot.c	2025-04-28 21:20:58.768046707 -0700
@@ -977,6 +977,10 @@
   r=next_chmod(path, mode);
   if(r&&(errno==EPERM))
     r=0;
+#ifdef EACCES
+  if(r&&(errno==EACCES)) /* setting SUID bit in OPENAFS raises this */
+    r=0;
+#endif
 #ifdef EFTYPE		/* available under FreeBSD kernel */
   if(r&&(errno==EFTYPE))
     r=0;
@@ -1058,6 +1062,11 @@
   int r;
   INT_STRUCT_STAT st;
 
+#ifdef LIBFAKEROOT_DEBUGGING
+  if (fakeroot_debug) {
+    fprintf(stderr, "fchmodat path %s\n", path);
+  }
+#endif /* LIBFAKEROOT_DEBUGGING */
   /* If AT_SYMLINK_NOFOLLOW is set in the fchownat call it should
      be when we stat it. */
   r=INT_NEXT_FSTATAT(dir_fd, path, &st, flags & (AT_SYMLINK_NOFOLLOW | AT_EMPTY_PATH | AT_NO_AUTOMOUNT));
@@ -1076,6 +1085,10 @@
   r=next_fchmodat(dir_fd, path, mode, flags);
   if(r&&(errno==EPERM))
     r=0;
+#ifdef EACCES
+  if(r&&(errno==EACCES)) /* setting SUID bit in OPENAFS raises this */
+    r=0;
+#endif
 #ifdef EFTYPE		/* available under FreeBSD kernel */
   if(r&&(errno==EFTYPE))
     r=0;
